apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go

```diff
@@ -271,14 +271,27 @@ func (api *CrossModelRelationsAPI) registerRemoteRelation(relation params.Regist
 	}
 	logger.Debugf("relation token %v exported for %v ", relation.RelationToken, localRel.Tag().Id())
 
-	// Export the local application from this model so we can tell the caller what the remote id is.
+	// Export the local offer from this model so we can tell the caller what the remote id is.
+	// The offer is exported as an application name since it models the behaviour of an application
+	// as far as the consuming side is concerned, and also needs to be unique.
+	// This allows > 1 offers off the one application to be made.
 	// NB we need to export the application last so that everything else is in place when the worker is
 	// woken up by the watcher.
-	token, err := api.st.ExportLocalEntity(names.NewApplicationTag(localApplicationName))
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return nil, errors.Annotatef(err, "exporting local application %v", localApplicationName)
+
+	// Juju 2.5.1 and earlier exported the local application name so for backwards compatibility
+	// use that if it's there.
+	token, err := api.st.GetToken(names.NewApplicationTag(localApplicationName))
```

Comment:
Will this do the right thing if there's an offer created  in 2.5.1 or earlier and then another offer for the same app is created in 2.5.2+? It seems like the new offer would also pick up the application-based token. Maybe converting back from token to offer and checking it's the one we expect would solve that?